### PR TITLE
feat: use cross-fetch instead of XMLHttpRequest to handle redirects

### LINKED ISSUE #14: did.json hosted under a redirect does not resolve
A quick debug points to the resolver getting stuck on [this line](https://github.com/decentralized-identity/web-did-resolver/blob/10008a2a20d9f690df40dcde7c0e062946c5af86/src/resolver.ts#L23) and never completing the promise.

At the time of writing this `did:web:uport.me` is experiencing this issue.

### PR DESCRIPTION
**Pull request by mi-xu**
This PR fixes https://github.com/decentralized-identity/web-did-resolver/issues/14

- Replaced `XMLHttpRequest` with `axios` for making Ajax requests.  This provides a higher level syntax for simpler and more robust handling of HTTP responses with a status other than 200.  It comes at the cost of an additional 1.8kb package dependency (gzipped + minified), which I feel is worth the tradeoff compared to reinventing the wheel for handling a lot of edge cases.

Test by registering the resolver to resolve "did:web:uport.me"
```
import { Resolver } from 'did-resolver'
import getResolver from 'web-did-resolver'

const resolver = new Resolver(getResolver())
const doc = resolver.resolve('did:web:uport.me')
console.log(doc)
```

@ajunge @mirceanis does this solution work for the issues you encountered?

### CODE REVIEW

**Comment by pelle** [comment]
Since `fetch` is now everywhere, I'd prefer to use one of the fetch compatibility layers. Such as (but not necessarily) https://www.npmjs.com/package/cross-fetch

File: .vscode/settings.json [hunk 1]
```diff
@@ -4,6 +4,6 @@
     "editor.tabSize": 2,
     "editor.detectIndentation": false,
     "[typescript]": {
-      "editor.formatOnSave": true
+      "editor.formatOnSave": false
```
Comment (mirceanis) [inline .vscode/settings.json]: @mi-xu was this setting causing trouble?

Comment (mi-xu) [reply .vscode/settings.json]: Yes, although it was actually due to a conflict between the `.prettierrc` being configured to format everything with trailing commas, while the `tslint.json` was configured to never have trailing commas. I've changed the configs to consistently avoid trailing commas and reverted this setting instead.

**Review by mirceanis — APPROVED** [review]
looks good to me
